packages/web-components/src/components/cbp-toggle/cbp-toggle.tsx

```diff
@@ -39,18 +39,64 @@ export class CbpToggle {
     });
   }
 
+  @Listen('keydown')
+  handleKeyDown( ev: KeyboardEvent){
+    if(ev.key === ' '){
+      ev.preventDefault();
+      this.toggleEvent();
+    }
+  }
+
+  @Event() toggleClick: EventEmitter;
+
   /** Event: toggles the control true/false & updates DOM accordingly*/
   toggleEvent(){
-    //TODO: logic here
+    let checkbox = this.host.querySelector('input[type="checkbox"]');
+
+    if(this.checked && !this.disabled){
+      this.checked = false;
+      checkbox.setAttribute('checked', 'false');
+    }else if(!this.disabled) {
+      this.checked = true;
+      checkbox.setAttribute('checked', 'true');
+    }
   }
 
   render() {
-    /**boilerplate HTML */
-    return (
-      <Host>
+    let ariaLabel = this.host.querySelector('label').textContent;
+    let ariaCheck = this.checked;
+
+    if(!this.hideStatus){
```

Comment:
This should be doable as 1 return with a logical and controlling the status text output.

```
{!this.hideStatus && <span>{this.checked ? this.statusTextOn : this.statusTextOff}</span>}
```

Reply:
updated as requested to consolidate